components/sceneutil/workqueue.cpp

```diff
@@ -32,6 +32,8 @@ WorkItem::WorkItem()
 
 WorkItem::~WorkItem()
 {
+    // Make sure that no one locked this work item.
+    OpenThreads::ScopedLock<OpenThreads::Mutex> lock(mMutex);
```

Comment:
Honestly, if an object is deleted when it's still in use and needs locking, there's a more fundamental issue regarding the object's lifetime and ownership. Something still has an access handle to it even though it's being freed. Work on an object should be done before it's deleted, not during.

Reply:
Actually, you use the same approach in the `OpenAL_Output::StreamThread` destructor in [this](https://github.com/OpenMW/openmw/commit/0f05ccf72a20b0a2a927509c58b30dabdb0520a0) commit.

Reply:
`OpenAL_Output` owns the thread the mutex is used for helping signal. The mutex is used to ensure the controlled thread knows to end before `join()` is called, in the same destructor. It's different here since this isn't being used to signal or join anything, but to temporarily halt deletion of the object while another handle in another thread is calling a method on it. But if another handle exists in another thread when deletion occurs, there's an ownership/lifetime problem with the object.